specs/flyout/flyout.md

```diff
@@ -0,0 +1,184 @@
+
+This includes:
+- Trap focus - optional, defaults to enabled - Focus is contained within the Flyout while it is visible.
+- First element focus - Focusing on the first focusable element when trap focus is enabled, but not overriding an element that is set to be focused by the author.
+- Returning focus to the target - after Flyout is closed the focus would return to the target element per the expectations of the Dialog design pattern.
+
+### Risks and Challenges
+Trap focus in the shadowDom has risk as it can be problematic and the FAST teams is currently re-evaluating how this is accomplished.
+For more insight: [Managing focus in the shadow DOM](https://nolanlawson.com/2021/02/13/managing-focus-in-the-shadow-dom/)
+
+
+### Prior Art/Examples
+
+#### Popover from Salesforce
+- https://www.lightningdesignsystem.com/components/popovers/
+
+![Popover](./images/salesforce-popover.png)
+#### Popover from Evergreen
+- triggered by a button
+- https://evergreen.segment.com/components/popover/
+
+![Popover](./images/evergreen-popover.png)
+
+#### Microsoft Windows Developer Flyout, Text Only
+![Text Only](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-wrapping-text.png)
+
+#### Microsoft Windows Developer Flyout, Actionable
+![Actionable More Info](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-example2.png)
+
+#### Inline Dialog from ATLAssian
+- https://atlassian.design/components/inline-dialog/examples
+
+![Popover](./images/atlassian-popover.png)
+
+#### Callout with actions from Fluent UI
+- https://developer.microsoft.com/en-us/fluentui#/controls/web/callout
+![Callout](./images/fluent-callout.png)
+
+
+---
+
+### API
+*Component Name*
+- `fast-flyout`
+
+*Attributes*
+- `hidden` - boolean, whether or not the flyout is hidden (defaults to false).
+- `trap-focus` - boolean, whether or not to keep focus contained inside the Flyout, defaults to true
+- `target` - string, the html id of the HTMLElement that the Flyout is attached to, (viewport for anchored region defaults to the parent of the anchor) and triggered by
+- `position` - enum, where the tooltip should appear relative to its target, uses Anchored Region logic, see *Position Options*
+- `delay` - number, milliseconds (ms) the time delay before the Flyout is shown once triggered
+- `aria-labelledby` - optional based on implementation**
+- `aria-describedby` - optional based on implementation**
+- `aria-label` - optional, based on implementation**
+
+    ** See the [W3C Specification](https://w3c.github.io/aria-practices/#dialog_roles_states_props) for requirements and details.
+
+*Properties*
+- `targetElement` - the reference to the HTMLElement used as target
+- `viewportElement` - the reference to the HTMLElement used as the viewport, required to set up anchored region properly
```

Comment:
This defaults to the document if unset in anchored region, so viewport can be left unset.  "target" is required.   You could probably have an attribute for viewport as well as target.

Reply:
Ok I'll swap the required statements.

Reply:
I rebased which outdated these comments :/ ... I have updated though so let me know what you think!